h2o-core/src/main/java/water/parser/ParseSetup.java

```diff
@@ -630,14 +598,15 @@ private static final long maxLineLength(byte[] bytes) {
         while(true) {
           line = br.readLine();
           if (line == null) break;
-          maxLineLength = Math.max(line.length(), maxLineLength);
+          int ll = line.length();
+          if (ll > maxLineLength) {
+            maxLineLength = ll;
+          }
         }
-      } catch (IOException e) {
-        return -1;
-      }
+      } catch (IOException ignore) {}
       return maxLineLength;
     }
-    return -1;
```

Comment:
An empty file contains exactly 0 bytes and 0 lines. That's the right answer. -1 is not the length of an empty file. You can try wc.
Craigs-MacBook-Pro-2:h2o-3 vpatryshev$ touch empty.txt
Craigs-MacBook-Pro-2:h2o-3 vpatryshev$ ls -l empty.txt
-rw-r--r--  1 vpatryshev  staff  0 Apr 11 14:26 empty.txt
Craigs-MacBook-Pro-2:h2o-3 vpatryshev$ wc empty.txt
       0       0       0 empty.txt
Craigs-MacBook-Pro-2:h2o-3 vpatryshev$